packages/bot-web-ui/src/components/quick-strategy/form.tsx

```diff
@@ -76,16 +83,26 @@ const QuickStrategyForm = observer(() => {
                         ) {
                             return null;
                         }
+
+                        const excluded_fields: Array<string> = ['last_digit_prediction', 'label_last_digit_prediction'];
+                        const excluded_trade_types: Array<string> = ['matchesdiffers', 'overunder'];
+
+                        if (
+                            excluded_fields.includes(field.name as string) &&
+                            values.type !== 'DIGITMATCH' &&
+                            !excluded_trade_types.includes(values.tradetype as string)
+                        ) {
+                            return null;
+                        }
```

Comment:
@maryia-matskevich-deriv do we still need this as we are already handling the visibility of the field using the `should_have` and `hide_without_should_have` flags from the config file? I think it would be better if we could avoid adding field-specific logic to the code as moving forward there could be more fields added. So maintaining everything from the config file would be helpful to keep the codebase clean.